### PR TITLE
decent-sampler: init at 1.9.4

### PR DESCRIPTION
**Pull request by adam248**
## Decent Sampler

https://www.decentsamples.com/product/decent-sampler-plugin/

Resolving Issue: https://github.com/NixOS/nixpkgs/issues/238499

This is my very first contribution to `nixpkgs` and I have tried to make it as simple as possible.
I have read as much as I can about how to do this correctly but if I missed something please let me know.
Thanks

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by tobiasBora** [comment]
Thanks, but a random questions: why do you use buildFHSEnv instead of autopatchelfhook? We usually try to avoid BuildFHS as it creates a sandbox that makes integration with other softwares more complicated, and in 90% of cases autopatchelfhook does a good enough job.

**Comment by adam248** [comment]
Thanks @tobiasBora , I have documented my journey trying to get this to work in this issue here:  https://github.com/NixOS/nixpkgs/issues/238499

I have tried autopatch and manually patching the binary but it always leads to a seg fault when running the program. This seems to match the buildFHS use case, as the docs for this function say it is for binaries that expect an FHS-compatible system.

The first time I got it working was with buildFHS, even though I know this is not the first choice, I just really want to get my first package done.

Right now the VST is not being made available for other programs and only the main binary is being packaged so you can at least run the base program directly.

The only thing I haven't tried is the `wrapProgram` or `makeWrapper` as I am still trying to learn how they work exactly.

I fully plan to rewrite this package in the future to include the VST library linking under `$out/share/{vst, vst3}` but for now I just want to make sure the main program works with this setup.

I fully intend to maintain and improve this package for the foreseeable future.

**Comment by adam248** [comment]
@IogaMaster just watched your video `HOW to REALLY learn NixOS` and I am taking you up on your offer to review my PR. thanks

File: pkgs/by-name/de/decent-sampler/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/de/decent-sampler/package.nix:32]:
```suggestion
  libs = [
```

File: pkgs/by-name/de/decent-sampler/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/de/decent-sampler/package.nix:8]:
```suggestion
, libX11
```

File: pkgs/by-name/de/decent-sampler/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/de/decent-sampler/package.nix:26]:
```suggestion
      runHook preInstall

```

File: pkgs/by-name/de/decent-sampler/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/de/decent-sampler/package.nix:27]:
```suggestion
      install -Dm755 DecentSampler -t $out/bin/decent-sampler

      runHook postInstall
```

File: pkgs/by-name/de/decent-sampler/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/de/decent-sampler/package.nix:29]:
```suggestion
```

File: pkgs/by-name/de/decent-sampler/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/de/decent-sampler/package.nix:22]:
```suggestion
      hash = "sha256-O/0R70tZOmSGgth6nzt4zPiJr1P8890uzk8PiQGnC6M=";
```

File: pkgs/by-name/de/decent-sampler/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/de/decent-sampler/package.nix:56]: This should be a code comment above `meta.licenses`, not in the desc

File: pkgs/by-name/de/decent-sampler/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/de/decent-sampler/package.nix:60]:
```suggestion
    license = licenses.unfree;
```
no license means it's unfree

File: pkgs/by-name/de/decent-sampler/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/de/decent-sampler/package.nix:61]:
```suggestion
    platforms = [ "x86_64-linux" ];
```

File: pkgs/by-name/de/decent-sampler/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/de/decent-sampler/package.nix:62]: please set `meta.mainProgram` to the binary's name

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

**Comment by eclairevoyant** [comment]
Also, as said above, FHS env is best avoided if possible. If we can wrap the program that would be much better, though I'll have to look further when I'm at a PC

File: pkgs/by-name/de/decent-sampler/package.nix
Comment (eclairevoyant) [reply pkgs/by-name/de/decent-sampler/package.nix:22]: borgo also identifies a hash mismatch here:

```
error: hash mismatch in fixed-output derivation '/nix/store/vf1z2cly6kzd9par02zfqbral6nf8rw4-source.drv':
         specified: sha256-O/0R70tZOmSGgth6nzt4zPiJr1P8890uzk8PiQGnC6M=
            got:    sha256-lTp/mukCwLNyeTcBT68eqa7aD0o11Bylbd93A5VCILU=
```

**Comment by eclairevoyant** [comment]
Finally, please rebase/squash into 2 commits: 1st one adding yourself to maintainers, 2nd with the new package, and ensure to follow contributing.md. And **do not merge master, always rebase**.

File: pkgs/by-name/de/decent-sampler/package.nix
Comment (PowerUser64) [reply pkgs/by-name/de/decent-sampler/package.nix:22]: I encountered this too.

Comment (adam248) [reply pkgs/by-name/de/decent-sampler/package.nix:22]: thanks guys, I just realized that I built it for version 1.9.8 first from dropbox, but then changed to archive.org for better reliability but forgot to reset the hash.

**Comment by adam248** [comment]
Thank you everyone for your reviews. I will work on fixing this up next week.

**Comment by IogaMaster** [comment]
Squash all the commits and remoive the merge commit from the history

**Comment by Janik-Haag** [comment]
I think you accidentally dropped all your commits you should be able to use git reflog to undo that and force push to the branch again then the pr should reopen.

**Comment by adam248** [comment]
Thanks for that @Janik-Haag. I am a bit new to rebasing, so I have been hitting my head against a brick wall.
I was getting an error when trying to push so I ended up creating a new branch and starting all over again.
But all I was missing was `FORCE` pushing the rebase to the PR!!! oh dear!!!

I am very sorry everyone for the mess! Please continue with this PR https://github.com/NixOS/nixpkgs/pull/275535

**Comment by PowerUser64** [comment]
Don't stress about it @adam248, we've all been there https://github.com/NixOS/nixpkgs/pull/242616#issuecomment-1628694947 :sweat_smile:

**Comment by Janik-Haag** [comment]
> Don't stress about it @adam248, we've all been there https://github.com/NixOS/nixpkgs/pull/242616#issuecomment-1628694947 😅

Yup same here :see_no_evil: 
In my personal workflow which might not work for you, I use `git commit --amend` which is a git feature that says hey this commit fixes something from the previous commit. And then you can do `git rebase -i --autosquash` and it automatically squashes all the amended commits into the original one. And if you want to learn more about git in a somewhat playfull way I can recommend https://ohmygit.org/ which is a open source game trying to teach git.

**Pull request closed without merging**
